### DISCUSSION TITLE
Too big dependencies of ultralytics

### DISCUSSION THREAD

**Discussion by quanqigu** [ultralytics/ultralytics#16008] [General]
When using ultralytics with triton, it actually no need torch or tlops, which make the docker image too bigger.

Is there anyway to reduce the decencies?
In ubuntu, it also install cuda related package, how can I remove such related dependencies?

**Comment by glenn-jocher** [comment]
@quanqigu to reduce dependencies, consider creating a custom environment by installing only the necessary components for your use case. You can manually install Ultralytics without optional dependencies and avoid installing CUDA-related packages if they are not needed.

**Comment by quanqigu** [comment]
@glenn-jocher I finally get the answer from this issue. https://github.com/ultralytics/ultralytics/issues/2542
Adding such magic index url will solve the problem, my docker image shrink from 3.2G to 700M.
```
--extra-index-url https://download.pytorch.org/whl/cpu
ultralytics
```

**Reply by glenn-jocher** [reply]
@quanqigu great to hear you found a solution! Using the CPU-only PyTorch index is a smart way to reduce the Docker image size. If you have any more questions, feel free to ask.
